service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/NotifyGatekeeperTest.java

```diff
@@ -79,15 +83,33 @@ void shouldBuildGatekeeperNotificationTemplateWithCompleteValues() throws Except
             .put("caseUrl", "http://fake-url/case/" + JURISDICTION + "/" + CASE_TYPE + "/12345")
             .build();
 
-        mockMvc
-            .perform(post("/callback/notify-gatekeeper/submitted")
+        makeRequest(callbackRequest(), "submitted");
+
+        verify(notificationClient, times(1)).sendEmail(GATEKEEPER_SUBMISSION_TEMPLATE, GATEKEEPER_EMAIL,
+            expectedGatekeeperParameters, "12345");
+    }
+
+    private MvcResult makeRequest(CallbackRequest request, String endpoint) throws Exception {
+        return mockMvc
+            .perform(post("/callback/notify-gatekeeper/" + endpoint)
                 .header("authorization", AUTH_TOKEN)
                 .header("user-id", USER_ID)
                 .contentType(MediaType.APPLICATION_JSON)
-                .content(readBytes("core-case-data-store-api/callback-request.json")))
-            .andExpect(status().isOk());
+                .content(mapper.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+    }
 
-        verify(notificationClient, times(1)).sendEmail(GATEKEEPER_SUBMISSION_TEMPLATE, GATEKEEPER_EMAIL,
-            expectedGatekeeperParameters, "12345");
+    private AboutToStartOrSubmitCallbackResponse requestWithoutCaseNumber() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
```

Comment:
I think i'd keep this method as 'buildCallbackRequest' that returns a CallbackRequest and move the rest of it out, lets discuss on slack